### PR TITLE
Fix cannot hide terminal find widget when pressing `esc`

### LINKED ISSUE #183087: Cannot hide terminal find widget when pressing `esc`
- VS Code Version: stable and from sources

Steps to Reproduce:

1. open terminal panel
3. split terminal
4. show find widget in both terminal
5. focus first terminal and press `esc` to hide find widget
6. :bug: focus second terminal and press `esc` to hide find widget, doesn't work

Regression from https://github.com/microsoft/vscode/pull/175785

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #183087

Before, find contribution was using the global contextService so context keys values were incorrect, convert to a proper terminal contribution so the terminal instance scoped contextService is used instead.

cc @Tyriar

### CODE REVIEW

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts [hunk 1]
```diff
@@ -3,65 +3,90 @@
...
+
+	override dispose() {
+		super.dispose();
 
-		findWidgets.set(instance, widget);
-		result = widget;
+		const focusTerminal = this._focusState;
+		this._findWidget.rawValue?.dispose();
+		if (focusTerminal) {
+			this._instance.focus();
```
Comment (jeanp413) [inline src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts:84]: Is this focus on dispose needed :thinking: ? it was in the original code

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 2]
```diff
@@ -163,8 +163,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	private _title: string = '';
 	private _titleSource: TitleEventSource = TitleEventSource.Process;
 	private _container: HTMLElement | undefined;
-	get container(): HTMLElement | undefined { return this._container; }
 	private _wrapperElement: (HTMLElement & { xterm?: XTermTerminal });
+	get domElement(): HTMLElement { return this._wrapperElement; }
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:167]: Could this be removed in favor of `xterm.raw.element` in `TerminalFIndContribution.xtermReady`?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts:84]: I guess not. I want terminalContribs to work similar to xterm.js addons where they can be enabled/disabled without the terminal disposing, but still it should be up to the terminal instance to restore focus in this case so I think we can safely remove this

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:167]: How about `xterm.raw.element.parentElement`? If not, anything beyond `parentElement` is too hacky

Comment (jeanp413) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:167]: Parent element is this [plain div here](https://github.com/microsoft/vscode/blob/70fa2da3646d385e6e07c834abe3f5b41eada4f7/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L899-L900C8) that can moved to be created in constructor too and exposed through the interface, also maybe is possible to remove it so the parent element is `_wrapperElement` itself
Or do you mean you prefer not to expose the dom element in the `terminalInstance` interface

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:167]: Any interfaces that make sense to remove from `ITerminalInstance` is a win, but we can leave it as it's not specifically fixing this issue.

**Review by Tyriar — APPROVED** [review]

**Review by andreamah — APPROVED** [review]

**Pull request merged**